### ISSUE TITLE
Realm settings override tokens results in error

### ISSUE THREAD

**Issue by edewit** [keycloak/keycloak#18680]
Go to realm settings and then to tokens at the bottom of the screen you can override the token lifespan. If you don't update / change all of them you keycloak crashes and can't startup again

![image](https://user-images.githubusercontent.com/51133/130313181-8841bf26-370a-46c7-8ad4-6db43610819f.png)

keycloak log file:
```
08:43:18,746 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (default task-4) Uncaught server error: java.lang.NumberFormatException: For input string: ""
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:662)
	at java.base/java.lang.Integer.valueOf(Integer.java:983)
	at org.keycloak.keycloak-model-jpa@15.0.2//org.keycloak.models.jpa.RealmAdapter.lambda$getUserActionTokenLifespans$2(RealmAdapter.java:579)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1746)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.keycloak.keycloak-model-jpa@15.0.2//org.keycloak.models.jpa.RealmAdapter.getUserActionTokenLifespans(RealmAdapter.java:579)
	at org.keycloak.keycloak-model-infinispan@15.0.2//org.keycloak.models.cache.infinispan.entities.CachedRealm.<init>(CachedRealm.java:242)
	at org.keycloak.keycloak-model-infinispan@15.0.2//org.keycloak.models.cache.infinispan.RealmCacheSession.getRealm(RealmCacheSession.java:414)
	at org.keycloak.keycloak-model-jpa@15.0.2//org.keycloak.models.jpa.JpaRealmProvider.getRealmByName(JpaRealmProvider.java:156)
	at org.keycloak.keycloak-model-infinispan@15.0.2//org.keycloak.models.cache.infinispan.RealmCacheSession.getRealmByName(RealmCacheSession.java:450)
	at org.keycloak.keycloak-services@15.0.2//org.keycloak.services.managers.RealmManager.getRealmByName(RealmManager.java:92)
	at org.keycloak.keycloak-services@15.0.2//org.keycloak.services.resources.admin.RealmsAdminResource.getRealmAdmin(RealmsAdminResource.java:170)
```

**Comment by devloe** [comment]
It still happens in the latest versions of Keycloak. In my case, I've created some realms using the API, but for some reason, the following values in the 'realm_attribute' of the database were created empty:

actionTokenGeneratedByUserLifespan
actionTokenGeneratedByUserLifespan-execute-actions
actionTokenGeneratedByUserLifespan.execute-actions
actionTokenGeneratedByUserLifespan.idp-verify-account-via-email
actionTokenGeneratedByUserLifespan.reset-credentials
actionTokenGeneratedByUserLifespan-verify-email

The action that triggered the error was changing the 'Default Token Algorithm' in the realm's settings to 'ES256'.

**Comment by sschu** [comment]
@devloe This is fixed in 23.0.6, see https://github.com/keycloak/keycloak/pull/26729.
